### PR TITLE
provider_resolver migration from provider_mapping

### PR DESCRIPTION
**Pull request by lamont-granquist**
- move macosx providers to dynamic resolution
- move windows providers to dynamic resolution
- move openbsd/netbsd providers to dynamic resolution
- move all cron providers to dynamic resolution
- move mdadm to static mapping

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@chef/client-core @chef/client-engineers needs review.

this is unfinished business from the provider resolver and moving towards the complete deprecation of the old platform_map.

**Comment by tyler-ball** [comment]
:+1: and :cake:

File: spec/unit/provider_resolver_spec.rb [hunk 1]
```diff
@@ -452,6 +452,137 @@ def stub_service_configs(*configs)
...
+
+
+        it "returns a Chef::Provider::Package::Homebrew provider" do
+          expect(resolved_provider).to eql(Chef::Provider::Package::Homebrew)
+        end
+      end
+    end
+  end
+
+  hash = {
```
Comment (jdmundrawala) [inline spec/unit/provider_resolver_spec.rb]: i really hate this name

**Comment by jdmundrawala** [comment]
:+1:

File: spec/unit/provider_resolver_spec.rb (hunk 1 above)
Comment (lamont-granquist) [inline spec/unit/provider_resolver_spec.rb]: lol, yeah, i meant to rename it after i figured out what it was...

**Pull request merged**